src/duckling/game/path/path-drawer.ts

```diff
@@ -0,0 +1,168 @@
+import {Graphics, DisplayObject, Container} from 'pixi.js';
+
+import {DrawnConstruct} from '../../canvas/drawing';
+import {Entity} from '../../entitysystem/entity';
+import {drawRectangle} from '../../canvas/drawing/util';
+import {Vector, vectorRotate} from '../../math/vector';
+
+import {getPath} from './path-attribute';
+
+export const PATH_HEIGHT = 4;
+const VERTEX_MARKER_HEIGHT = 2 * PATH_HEIGHT;
+const TRIANGLE_HEIGHT = 2 * VERTEX_MARKER_HEIGHT;
+const DIRECTION_INDICATOR_OFFSET = VERTEX_MARKER_HEIGHT / 2;
+const JOINT_HEIGHT = VERTEX_MARKER_HEIGHT;
+const PATH_COLOR = 0x607d8b;
+const STARTING_SQUARE_COLOR = 0x00a626;
+const ENDING_SQUARE_COLOR = 0xf44336;
+
+export function drawPathAttribute(entity : Entity) : DrawnConstruct {
+    let pathAttribute = getPath(entity);
+    if (!pathAttribute || pathAttribute.vertices.length === 0) {
+        return null;
+    }
+
+    if (pathAttribute.vertices.length === 1) {
+        return _drawJoint(0, 0);
+    }
+    
+    if (pathAttribute.vertices.length === 2 && !pathAttribute.isLoop) {
+        return _drawSegmentWithBothSquares(pathAttribute.vertices[0], pathAttribute.vertices[1]);
+    }
+    
+    if (!pathAttribute.isLoop) {
+        return _drawPath(pathAttribute.vertices);
+    } else {
+        return _drawLoopPath(pathAttribute.vertices);
+    }
+}
+
+function _drawLoopPath(vertices : Vector[]) : DisplayObject {
+    let container = new Container();
+    for (let i = 1; i < vertices.length - 1; i++) {
```

Comment:
except we're leaving out the first segment to draw later when we call drawFirstLoopSegment